java/kafka/consumer/src/main/java/KafkaConsumerConfig.java

```diff
@@ -3,228 +3,57 @@
-            props.put(SaslConfigs.SASL_MECHANISM, "OAUTHBEARER");
-            if (!(additionalProps.containsKey(SaslConfigs.SASL_MECHANISM) && additionalProps.getProperty(SaslConfigs.SASL_MECHANISM).equals("PLAIN"))) {
-                props.put(SaslConfigs.SASL_LOGIN_CALLBACK_HANDLER_CLASS, config.saslLoginCallbackClass);
-            }
-        }
-
-        // override properties with defined additional properties
-        props.putAll(additionalProps);
-
-        return props;
+        String topic = System.getenv("STRIMZI_TOPIC");
+        Long messageCount = System.getenv("STRIMZI_MESSAGE_COUNT") == null ? DEFAULT_MESSAGES_COUNT : Long.parseLong(System.getenv("STRIMZI_MESSAGE_COUNT"));
+        TracingSystem tracingSystem = TracingSystem.forValue(System.getenv().getOrDefault("STRIMZI_TRACING_SYSTEM", ""));
+        Properties properties = new Properties();
+        properties.putAll(System.getenv()
+                .entrySet()
+                .stream()
+                .filter(mapEntry -> mapEntry.getKey().startsWith(KAFKA_PREFIX))
+                .collect(Collectors.toMap(mapEntry -> convertEnvVarToPropertyKey(mapEntry.getKey()), Map.Entry::getValue)));
+        return new KafkaConsumerConfig(topic, messageCount, tracingSystem, properties);
     }
 
-    public String getBootstrapServers() {
-        return bootstrapServers;
+    private static String convertEnvVarToPropertyKey(String envVar) {
+        return envVar.substring(envVar.indexOf("_") + 1).toLowerCase().replace("_", ".");
     }
 
     public String getTopic() {
         return topic;
     }
 
-    public String getGroupId() {
-        return groupId;
-    }
-
-    public String getAutoOffsetReset() {
-        return autoOffsetReset;
-    }
-
-    public String getEnableAutoCommit() {
-        return enableAutoCommit;
-    }
-
-    public String getClientRack() {
-        return clientRack;
-    }
-
     public Long getMessageCount() {
         return messageCount;
     }
 
-    public String getSslTruststoreCertificates() {
-        return sslTruststoreCertificates;
-    }
-
-    public String getSslKeystoreKey() {
-        return sslKeystoreKey;
-    }
-
-    public String getSslKeystoreCertificateChain() {
-        return sslKeystoreCertificateChain;
-    }
-
-    public String getOauthClientId() {
-        return oauthClientId;
-    }
-
-    public String getOauthClientSecret() {
-        return oauthClientSecret;
+    public TracingSystem getTracingSystem() {
+        return tracingSystem;
     }
 
-    public String getOauthAccessToken() {
-        return oauthAccessToken;
-    }
-
-    public String getOauthRefreshToken() {
-        return oauthRefreshToken;
-    }
-
-    public String getOauthTokenEndpointUri() {
-        return oauthTokenEndpointUri;
-    }
-
-    public String getAdditionalConfig() {
-        return additionalConfig;
-    }
-
-    public TracingSystem getTracingSystem() { return tracingSystem; }
-
-    @Override
-    public String toString() {
```

Comment:
instead of deleting this one, doesn't it make more sense to print fields like topic, messageCount and so on, by adding the properties field as well. I think it's useful to allow logging the consumer configuration.
@scholzj wdyt?

Reply:
Yeah, sure. That would make sense.